test/kitchen/site-cookbooks/dd-security-agent-check/recipes/functional-tests.rb

```diff
@@ -0,0 +1,273 @@
+      action :create
+      setup_docker_repo false
+      package_name 'docker'
+      package_options %q|-y|
+    end
+
+    service 'docker' do
+      action [ :enable, :start ]
+    end
+  elsif ['ubuntu'].include?(node[:platform])
+    docker_installation_package 'default' do
+      action :create
+      setup_docker_repo false
+      package_name 'docker.io'
+    end
+  else
+    docker_service 'default' do
+      action [:create, :start]
+    end
+  end
+
+  if node[:cws_platform] == "docker"
+    # Please see https://github.com/paulcacheux/cws-buildimages/blob/main/Dockerfile
+    # for the definition of this base image.
+    # If this successfully helps in reducing the amount of rate limits, this should be moved
+    # to DataDog/datadog-agent-buildimages.
+    file "#{node['common']['work_dir']}/Dockerfile" do
+      content <<-EOF
+      FROM ghcr.io/paulcacheux/cws-centos7@sha256:4fc1aac178b5c1690ce71c37f22b8a23cedfb969c7056702c21be50e848e554f
+
+      ADD nikos.tar.gz /opt/datadog-agent/embedded/nikos/embedded/
+
+      COPY clang-bpf /opt/datadog-agent/embedded/bin/
+      COPY llc-bpf /opt/datadog-agent/embedded/bin/
+
+      CMD sleep 7200
+      EOF
+      action :create
+    end
+
+    docker_image 'testsuite-img' do
+      tag 'latest'
+      source node['common']['work_dir']
+      action :build
+    end
+
+    docker_container 'docker-testsuite' do
+      repo 'testsuite-img'
+      tag 'latest'
+      cap_add ['SYS_ADMIN', 'SYS_RESOURCE', 'SYS_PTRACE', 'NET_ADMIN', 'IPC_LOCK', 'ALL']
+      volumes [
+        # security-agent misc
+        '/tmp/security-agent:/tmp/security-agent',
+        # HOST_* paths
+        '/proc:/host/proc',
+        '/etc:/host/etc',
+        '/sys:/host/sys',
+        # os-release
+        '/etc/os-release:/host/etc/os-release',
+        '/usr/lib/os-release:/host/usr/lib/os-release',
+        # passwd and groups
+        '/etc/passwd:/etc/passwd',
+        '/etc/group:/etc/group',
+      ]
+      env [
+        'HOST_PROC=/host/proc',
+        'HOST_ETC=/host/etc',
+        'HOST_SYS=/host/sys',
+      ]
+      privileged true
+    end
+
+    docker_exec 'debug_fs' do
+      container 'docker-testsuite'
+      command ['mount', '-t', 'debugfs', 'none', '/sys/kernel/debug']
+    end
+
+    for i in 0..7 do
+      docker_exec 'create_loop' do
+        container 'docker-testsuite'
+        command ['bash', '-c', "mknod /dev/loop#{i} b 7 #{i} || true"]
+      end
+    end
+  end
+end
+
+if platform_family?('centos', 'fedora', 'rhel')
+  selinux_state "SELinux Permissive" do
+    action :permissive
+  end
+end
+
+if File.exists?('/sys/kernel/security/lockdown')
+  file '/sys/kernel/security/lockdown' do
+    action :create_if_missing
+    content "integrity"
+  end
+end
+
+# system-probe common
+
+system_probe_tests_folder = '/tmp/system-probe-tests'
+
+directory system_probe_tests_folder do
+  recursive true
+end
+
+file "#{system_probe_tests_folder}/color_idx" do
```

Comment:
FYI this path is changing in https://github.com/DataDog/datadog-agent/pull/15670 so if that gets merged first, you should update this path.